File: javascript-exercises-master/calculator/calculator.js
function add (n1,n2) {
return n1+n2
	
}

function subtract (n1,n2) {
	return n1-n2
}

function sum (x) {
 	const totalx=x.reduce((total,x)=>{return total+ x},0);
   return totalx;
}

function multiply (x) {

	const totalx=x.reduce((total,x)=>{return total* x},1);
   return totalx;
}

function power(n1,n2) {
	let total=1;
 for (i=1;i<=n2;i++)
 	{total*=n1}
 return total;
}

function factorial(n1) {
	let total=1;
	if (n1==0) {return 1}
	else {
		for(i=n1;i>0;i--){total*=i}
		return total;
	}
}

module.exports = {
	add,
	subtract,
	sum,
	multiply,
    power,
	factorial
}

File: learn_ruby-master/02_calculator/calculator.rb
def add (n1,n2)
 return n1+n2
end

def subtract (n1,n2)
	return n1-n2
end

def sum(array)
 sum = 0 
 array.each do|number|
 	 sum+= number
 end
 return sum
end

def multiply(array)
	mum = 0
	array.each do|number|
		mum*= number
	end
return mum
end

def power (n1,n2)
	total = 1
	n2.times do
	 total*=n1
	end
return total
end

def factorial (n1)
    total = 1
	if n1 == 0 
		total = 1
	else
		i = n1
		loop do 
		 total*=i
		 i-=1
		 break if (i==0)
		end
	end
return total
end